portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -86,12 +72,59 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         response.sendRedirect("/contact.html");
     }
 
+    /** Returns comment limit value from the request, or -1 if user selected 'all' */
+    private int getCommentLimit(HttpServletRequest request) {
+        String limit_string = request.getParameter("comment-limit");
+        int comment_limit = 0;
+
+        try {
+            comment_limit = Integer.parseInt(limit_string);
+        } catch (NumberFormatException n) {
+            if (limit_string.equals("all")) {
+                comment_limit = -1;
+            }
+            else {
+                // Printing error messages?
```

Comment:
This looks like an internal error, as the html page should not return a value you don't expect.
It looks like the proper way to handle this would be for getCommentLimit() to throw an exception, and then have the servlet return an error response.

But I suspect that sort of handling hasn't been covered yet.

Reply:
Got it, I will leave as is for now